src/app/k-tools/k-tools.component.html

```diff
@@ -1,12 +1,79 @@
-<ul class="list-group">
-  <a
-    class="list-group-item"
-    style="cursor: pointer;"
-    *ngFor="let ktool of kTools"
-  >
-    {{ ktool.name }} ({{ ktool.amount }})
-  </a>
-</ul>
-
-<mat-divider></mat-divider>
-<app-k-tool-edit></app-k-tool-edit>
+<div class="page-header">
+  <h1>External Koala Tools</h1>
+  <p>
+    Nulla labore exercitation id irure incididunt id cupidatat labore sit
+    aliquip sit fugiat. Nisi nulla qui aliqua ullamco officia commodo nostrud ad
+    Lorem excepteur commodo amet adipisicing cillum. Veniam deserunt dolor velit
+    ut amet. Mollit minim ex aliquip ex sit anim tempor velit irure proident
+    velit. Minim magna exercitation eu ullamco Lorem pariatur magna voluptate
+    enim dolor. Non sunt cillum Lorem eu proident. Ex dolore excepteur commodo
+    irure dolor fugiat duis adipisicing elit commodo commodo id nostrud id.
+  </p>
+</div>
+<div class="bd-highlight mb-3">
+  <div class="k-body shadow m-3 p-3 mb-5 bg-light rounded">
+    <!-- Filter/Sort Toolbar -->
+    <div class="mb-3">
+      <!-- Text Filter -->
+      <mat-form-field>
+        <mat-label>Filter</mat-label>
+        <input type="text" matInput [(ngModel)]="filterText" />
+      </mat-form-field>
+
+      <!-- Sort By -->
+      <mat-form-field class="ml-3">
+        <mat-label>Sort By</mat-label>
+        <mat-select [(ngModel)]="sortBy">
+          <mat-option value="name">Name</mat-option>
+          <mat-option value="numUsers">Users</mat-option>
+        </mat-select>
+      </mat-form-field>
+
+      <!-- Sort Direction -->
+      <mat-form-field class="ml-3">
+        <mat-label>Direction</mat-label>
+        <mat-select [(ngModel)]="direction">
+          <mat-option value="asc">Ascending</mat-option>
+          <mat-option value="dsc">Descending</mat-option>
+        </mat-select>
+      </mat-form-field>
+
+      <!-- Toggle Only Starred -->
+      <mat-checkbox class="ml-3" [(ngModel)]="onlyStarred">
+        Only Starred
+      </mat-checkbox>
+    </div>
+
+    <!-- Card Area -->
+    <div fxLayout="row rap">
```

Comment:
should this be row wrap?

Reply:
Strangely, it works with either "rap" or "wrap" lol. But I'll change it to the proper spelling.